### PR TITLE
[API][Product] Add subscriber for generating slug automatically

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | partially #11250 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/EventSubscriber/ProductSlugEventSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+{
+    /** @var SlugGeneratorInterface */
+    private $slugGenerator;
+
+    public function __construct(SlugGeneratorInterface $slugGenerator)
+    {
+        $this->slugGenerator = $slugGenerator;
+    }
+
+    public static function getSubscribedEvents()
```
Comment (pamil) [inline src/Sylius/Bundle/ApiBundle/EventSubscriber/ProductSlugEventSubscriber.php:35]:
```suggestion
    public static function getSubscribedEvents(): array
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/services.xml [hunk 2]
```diff
@@ -13,8 +13,9 @@
 
 <container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <imports>
-        <import resource="services/filters.xml" />
         <import resource="services/context_builders.xml" />
+        <import resource="services/event_subscribers.xml"/>
+        <import resource="services/filters.xml"/>
```
Comment (pamil) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services.xml:18]: Alphabetical order <3

File: src/Sylius/Bundle/ApiBundle/EventSubscriber/ProductSlugEventSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            KernelEvents::VIEW => ['generateSlug', EventPriorities::PRE_VALIDATE],
+        ];
+    }
+
+    public function generateSlug(ViewEvent $event): void
+    {
+        $product = $event->getControllerResult();
+        $method = $event->getRequest()->getMethod();
+
+        if (!$product instanceof ProductInterface || !in_array($method, [Request::METHOD_POST, Request::METHOD_PUT])) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/EventSubscriber/ProductSlugEventSubscriber.php:47]:
```suggestion
        if (!$product instanceof ProductInterface || !in_array($method, [Request::METHOD_POST, Request::METHOD_PUT], true)) {
```

Also, can we put it into multiple lines?

**Comment by lchrusciel** [comment]
Thanks, Grzegorz! :tada:

**Pull request merged**